### PR TITLE
Add fetch to see if patient has an existing patient method, if availa…

### PR DESCRIPTION
**Pull request by lawyu89**
…ble allow patient to use existing method to pay or enter a different cc payment

for sc
[sc-7621]

### CODE REVIEW

File: src/pages/Subscriptions.tsx [hunk 1]
```diff
@@ -148,66 +148,80 @@ const StripeErrorMessage = ({ message } : { message: string }) => (
...
     selectedSubscription,
-    setPageView
+    setPageView,
+    currentPatient,
+    existingPaymentMethod,
   } : {
     selectedSubscription: SubscriptionOption,
-    setPageView: (view: string) => void
+    setPageView: (view: string) => void,
+    currentPatient: any,
```
Comment (lawyu89) [inline src/pages/Subscriptions.tsx:157]: ive commited a cardinal sin 🤣

Comment (LUMacKenn) [reply src/pages/Subscriptions.tsx:157]: 🚨🚨🚨
📣 _come out with your hands up_
📣 _@ysael is on his way here now, do not resist arrest_

File: src/pages/Subscriptions.tsx [hunk 2]
```diff
@@ -148,66 +148,80 @@ const StripeErrorMessage = ({ message } : { message: string }) => (
...
-          subscription: {
-            patient_id: currentPatient.id,
-            price_ids: [selectedSubscription.id]
+      const { error, setupIntent } = await stripe.confirmCardSetup(
+        secret,
+        {
+          payment_method: {
+            card: elements.getElement(CardNumberElement),
+            billing_details: {
+              email: `${currentPatient.email}`,
```
Comment (LUMacKenn) [inline src/pages/Subscriptions.tsx:205]: are these values not strings already? If they are, let's remove the interpolation!

File: src/pages/Subscriptions.tsx [hunk 3]
```diff
@@ -148,66 +148,80 @@ const StripeErrorMessage = ({ message } : { message: string }) => (
...
+        setStripeError(error.message)
+        setLoading(false);
+      } else if (setupIntent.status === 'succeeded') {
+        await createSubscription()
+        setLoading(false);
+        setPageView('PaymentSuccess')
       }
-
+    } else {
+      await createSubscription()
```
Comment (LUMacKenn) [inline src/pages/Subscriptions.tsx:224]: Should we catch here (and on line 219), or wrap it all in a try catch maybe?

File: src/pages/Subscriptions.tsx [hunk 4]
```diff
@@ -216,25 +230,15 @@ const SubscriptionPayment = (
...
     }
 
     const { error, paymentIntent } = await stripe.confirmCardPayment(
       secret,
       {
-        payment_method: {
+        payment_method: existingPaymentMethod && !useNewCard ? existingPaymentMethod : {
           card: elements.getElement(CardNumberElement),
           billing_details: {
             email: `${currentPatient.email}`,
```
Comment (LUMacKenn) [inline src/pages/Subscriptions.tsx:244]: same question here about strings

File: src/pages/Subscriptions.tsx [hunk 5]
```diff
@@ -281,56 +285,77 @@ const SubscriptionPayment = (
...
+              width="100%"
+              type="submit"
+              variation="primary"
+              disabled={!stripe || !elements || !currentPatient || loading}
+              onClick={handleSubmit}
+            >
+              { loading ? <Loader /> : "Pay" }
+            </Button>
+            <Box sx={{ padding: "1rem", textAlign: "center" }}>
+              {/*@ts-ignore*/}
```
Comment (LUMacKenn) [inline src/pages/Subscriptions.tsx:303]: what was the ignore for here?

**Review by LUMacKenn — APPROVED** [review]
I haven't tested, and I don't know exactly how all the logic + edge cases _should_ work from the product/design perspective, but the code seems pretty good to me! Just left a couple minor comments but will approve now in case you wanna get it in tonight. I can test it tomorrow if you want

File: src/pages/Subscriptions.tsx (hunk 5 above)
Comment (lawyu89) [reply src/pages/Subscriptions.tsx:303]: Typography doesnt like variant value of eyebrow 🤨

File: src/pages/Subscriptions.tsx (hunk 4 above)
Comment (lawyu89) [reply src/pages/Subscriptions.tsx:244]: Youre right!

File: src/pages/Subscriptions.tsx (hunk 3 above)
Comment (lawyu89) [reply src/pages/Subscriptions.tsx:224]: Good call ill fix that

File: src/pages/Subscriptions.tsx (hunk 1 above)
Comment (ezekielp) [reply src/pages/Subscriptions.tsx:158]: 😂  🚓 👮
